PyInstaller/utils/hooks/__init__.py

```diff
@@ -1053,65 +1052,35 @@ def get_installer(module: str):
     return None
 
 
-# ``_map_distribution_to_packages`` is expensive. Compute it when used, then return the memoized value. This is a simple
-# alternative to ``functools.lru_cache``.
-def _memoize(f):
-    memo = []
+def requirements_for_package(package_name: str):
+    hiddenimports = []
 
-    def helper():
-        if not memo:
-            memo.append(f())
-        return memo[0]
+    # Helpers for retrieving top-level imports for a dist, based on `importlib.metadata` code.
+    def _top_level_declared(dist):
+        return (dist.read_text('top_level.txt') or '').split()
 
-    return helper
+    def _top_level_inferred(dist):
+        return [
+            f.parts[0] if len(f.parts) > 1 else f.with_suffix('').name for f in (dist.files or []) if f.suffix == ".py"
+        ]
```

Comment:
Could this mapping of distributions to modules not also be done by looking at `importlib.metadata.packages_distributions()`?

Reply:
It could, although `importlib.metadata.packages_distributions()` is exactly the opposite of what we need here (we'd need to invert the dictionary).

I think the only known caller of this function is `collect_all`, which needs `importlib.metadata.packages_distributions()` dictionary anyway; so it might make more sense to also invert it and perform the lookup there.

Or perhaps even drop this module -> dist -> requirements -> top-level-modules lookup  altogether.

Reply:
I've been meaning to scrap the requirements component of `collect_all()` for ages.

Reply:
Let's get rid of it here, then.